src/html/js/recommend.js

```diff
@@ -92,6 +100,19 @@ function getRecommendList(nowDate, callback) {
   })
 }
 
+// Filter planlist by keywords
+function filterByKeywords(planList, keywords) {
+  return _.filter(planList, function(event) {
+    if (keywords !== undefined && keywords.length > 0) {
+      return _.some(keywords, function(keyword) {
+        return _.contains(event.keywords, keyword);
+      });
+    } else {
+      return true;
```

Comment:
If there are no keywords, it is better not to return list.
planList is processed in duplicate.

Reply:
Thanks for your comment. I think you're right, so I modified it.
I added setRecommendSchedule(recommendSchedule, planList) later, so I forgot consideration in case of no keywords.